sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -518,10 +449,10 @@ object IntervalUtils {
               isNegative = false
             case '.' =>
               isNegative = false
-              fractionScale = (NANOS_PER_SECOND / 10).toInt
               i += 1
+              fractionScale = (NANOS_PER_SECOND / 10).toInt
               state = VALUE_FRACTIONAL_PART
-            case _ => return null
+            case _ => throwIAE( s"unrecognized sign '$nextWord'")
```

Comment:
we are trying to parse a number here, like `unrecognized number: '$nextword'`

Reply:
checkFromInvalidString("~1 hour", "unrecognized sign")

Reply:
number seems ok to me